### PR TITLE
feat: add mergePaths

### PR DESCRIPTION
**Pull request by BjornMelgaard**

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -350,6 +350,12 @@ declare namespace RamdaAdjunct {
         mergeProps(ps: Array<string>): (obj: Object) => Object;
 
         /**
+         * Merge objects under corresponting paths.
+         */
+        mergePaths(ps: Array<Array<string>>, obj: Object): Object;
```
Comment (char0n) [inline src/index.d.ts]: Signature of paths is 
```typescript
paths(ps: Array<Array<string | number>>, obj: Object): Array<any>;
```
They can also process arrays indexes that are numbers. Let's mirror it into mergePaths also

File: src/mergePaths.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+/**
+ * Merge objects under corresponting paths.
+ *
+ * @func mergePaths
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/1.17.0|v1.17.0}
+ * @category Object
+ * @sig [[k]] -> {k: {a}} -> {a}
+ * @see {@link RA.mergeProps|mergeProps}
+ * @param {!Array} ps The property paths to merge
```
Comment (char0n) [inline src/mergePaths.js]: Whould go for `The paths to merge`

File: src/mergePaths.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * @param {!Object} obj The object to query
+ * @return {!Object} The object composed of merged properties of obj
```
Comment (char0n) [inline src/mergePaths.js]: Would go for `The object composed of merged paths of obj`

File: src/mergePaths.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ * @example
+ *
+ * const obj = {
+ *   foo: { fooInner: { fooInner2: 1 } },
+ *   bar: { barInner: 2 }
+ * };
+ *
+ * R.mergePaths(
```
Comment (char0n) [inline src/mergePaths.js]: Can we go for more compact onel-iner here ?

```javascript
RA.mergePaths([['foo', 'fooInner'], ['bar']], obj); //=> { fooInner2: 1, barInner: 2 }
```

Also not `R` vs `RA` mismatch. Should be `RA`.

File: test/mergePaths.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    };
+    expected = { fooinner2: 1, barinner: 2 };
+  });
+
+  it('tests currying', function () {
+    eq(RA.mergePaths([['foo', 'fooinner'], ['bar']], obj), expected);
+    eq(RA.mergePaths([['foo', 'fooinner'], ['bar']])(obj), expected);
+  });
+
+  it('tests merging the props containing objects', function() {
```
Comment (char0n) [inline test/mergePaths.js]: Would go for `tests merging the paths containing object`

File: test/mergePaths.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    eq(RA.mergePaths([['foo', 'fooinner'], ['bar']], obj), expected);
+  });
+
+  it('tests merging the props containing non-objects', function () {
```
Comment (char0n) [inline test/mergePaths.js]: Would go for `tests merging the paths containing non-objects

**Comment by char0n** [comment]
Pls check the comments. Then we're good for merge & release.

File: src/index.d.ts (hunk 1 above)
Comment (BjornMelgaard) [reply src/index.d.ts]: hm, really, objects can be in arrays, neat

**Comment by BjornMelgaard** [comment]
Done)

File: src/index.d.ts [hunk 7]
```diff
@@ -350,6 +350,12 @@ declare namespace RamdaAdjunct {
         mergeProps(ps: Array<string>): (obj: Object) => Object;
 
         /**
+         * Merge objects under corresponting paths.
```
Comment (char0n) [inline src/index.d.ts]: corresponting -> corresponding

File: src/mergePaths.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
+import { curry, mergeAll } from 'ramda';
+
+import paths from './paths';
+
+/**
+ * Merge objects under corresponting paths.
```
Comment (char0n) [inline src/mergePaths.js]: corresponting -> corresponding

**Comment by BjornMelgaard** [comment]
dddone

**Comment by char0n** [comment]
Thanks for patience ;]

Cool tomorrow evening I'll make the v1.17.0 release.

**Pull request merged**